### PR TITLE
Rename return types for Datalake and file services.

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

**Comment by katmsft** [comment]
Also fixes https://github.com/Azure/azure-sdk-for-cpp/issues/461.

File: sdk/storage/inc/datalake/datalake_responses.hpp [hunk 1]
```diff
@@ -10,26 +10,27 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
   };
 
-  using FileSystemInfo = FileSystemCreateResponse;
+  using CreateFileSystemResult = FileSystemCreateResult;
+  using SetFileSystemMetadataResult = FileSystemCreateResult;
 
   // PathClient models:
 
-  using PathDeleteInfo = PathDeleteResponse;
+  using DeletePathResult = PathDeleteResult;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/datalake/datalake_responses.hpp:33]: what? cannot you just name in `DeletePathResult` in protocol layer?

File: sdk/storage/inc/shares/share_responses.hpp [hunk 2]
```diff
@@ -9,19 +9,21 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
+  using DeleteShareResult = ShareDeleteResult;
+  using CreateShareSnapshotResult = ShareCreateSnapshotResult;
+  using GetSharePropertiesResult = ShareGetPropertiesResult;
+  using SetShareQuotaResult = ShareSetQuotaResult;
+  using SetShareMetadataResult = ShareSetMetadataResult;
+  using SetShareAccessPolicyResult = ShareSetAccessPolicyResult;
+  using GetShareStatisticsResult = ShareGetStatisticsResult;
+  using CreateSharePermissionResult = ShareCreatePermissionResult;
+  using GetShareAccessPolicyResult = ShareGetAccessPolicyResult;
+  using GetSharePermissionResult = ShareGetPermissionResult;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/share_responses.hpp:27]: I feel awful about these.

File: sdk/storage/inc/datalake/datalake_responses.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/inc/datalake/datalake_responses.hpp:33]: I cannot.
E.g. for `ShareCreateSnapshotResult`, it comes from 3 parts: `Share` is the service name, `CreateSnapshot` is what swagger defines as the method's name and `Result` is appended by the generator.
If I changed the protocol layer to `DeletePathResult`, then `ShareCreateSnapshotResult` will be `CreateSnapshotShareResult`, which is way worse than the current name.

File: sdk/storage/inc/shares/share_responses.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/inc/shares/share_responses.hpp:27]: See the other comments.

File: sdk/storage/inc/datalake/datalake_responses.hpp (hunk 1 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/datalake/datalake_responses.hpp:33]: @JeffreyRichter said this naming rule only applies to convenience layer. So I guess we can name protocol layer variables whatever we like.

**Pull request merged**